lib/reporters/base.js

```diff
@@ -27,6 +27,11 @@ exports = module.exports = Base;
 
 var isatty = tty.isatty(1) && tty.isatty(2);
 
+/**
+ * Save log references to avoid tests interfering (see GH-3604).
+ */
+var consoleLog = console.log;
```

Comment:
IMO the "more correct" thing to do is to modify the eslint configuration to handle `console.*` the same way it handles timers (unless that's not desirable, for some reason?)

don't think we need to block on that, though; we can change it later.

Reply:
Had a quick look at this. The main issue being that outside the reporters we use `console` (log and error) a fair amount. The solution was always focused on fixing the issue with reporters. We could have a rule just for reporters (seems brittle), or we can completely this solution to follow the timers practice, but AFAIK there has been no indication we actually need this.

Reply:
the rules that exist for timers are also specific to certain files, so there’s a precedent for this anyway. again, not a blocker.

Reply:
Ive added the lint rule now.